src/tasks/LibraryBuilder/LibraryBuilder.cs

```diff
@@ -226,6 +253,79 @@ private static void WriteLinkerScriptFile(string exportsFile, List<string> expor
                 .Replace("%GLOBAL_SYMBOLS%", globalExports));
     }
 
+    private void WriteAutoInitializationFromTemplate()
+    {
+        string appContextEnvVariables = GenerateAppContextEnvVariables();
+        string assembliesLoader = GenerateAssembliesLoader();
+        string assetsPath = GenerateAssetsPath();
+        string runtimeConfig = GenerateRuntimeConfig();
+        File.WriteAllText(Path.Combine(OutputDirectory, "autoinit.c"),
+            Utils.GetEmbeddedResource("autoinit.c")
+                .Replace("%APPCTX_ENV_VARIABLES%", appContextEnvVariables)
+                .Replace("%ASSEMBLIES_LOADER%", assembliesLoader)
+                .Replace("%ASSETS_PATH%", assetsPath)
+                .Replace("%RUNTIME_CONFIG%", runtimeConfig));
+    }
+
+    private string GenerateAssembliesLoader()
+    {
+        var assembliesLoader = new StringBuilder();
+        foreach (string exportedAssembly in exportedAssemblies)
+        {
+            assembliesLoader.AppendLine($"    mono_assembly_open(\"{exportedAssembly}\", NULL);");
+        }
+        return assembliesLoader.ToString();
+    }
+
+    private string GenerateAssetsPath()
+    {
+        return AssetsPath ?? "DOTNET_ASSETS_PATH";
+    }
+
+    private string GenerateRuntimeConfig()
+    {
+        if (string.IsNullOrEmpty(RuntimeConfigBinFile))
+            return "    return;";
+
+        var runtimeConfig = new StringBuilder();
+        runtimeConfig.Append($"    char *file_name = {RuntimeConfigBinFile};");
```

Comment:
I'm not sure why this is done here?

Reply:
We should confirm if we always get a runtimeconfig file. I can't remember if we do or not.

Reply:
Right, I added this for now because I'm not familiar with that file and what it does, so added this for flexibility for now.

Reply:
I think we can assume that runtimeconfig.bin will be next to the assemblies, so I think this can be moved back into autoinit.c.  

I think we are going to have to run the `_AndroidGenerateRuntimeConfig` target before the library builder runs to make sure we have a bin file. The appbuilder targets do this currently because it's assumed they are building every piece of the app.

Reply:
I moved the `monovm_runtimeconfig_initialize` logic and `monovm_initialize` logic back into `autoinit.c`, with the only substitution being the appctx value for `RUNTIME_IDENTIFIER`, which has `RuntimeIdentifier` passed to it. 

Both of these will rely on the path to where assemblies are located (currently `bundle_path`) being set, which is done at the beginning of the runtime initialization callback.

The runtimeconfig file that is parsed in `monovm_runtimeconfig_initialize` has consistenly been named `runtimeconfig.bin`. Unless we anticipate the naming to vary (Android/Apple/Wasm/Wasi have it as `runtimeconfig.bin`) I dont think we *need* the `_AndroidGenerateRuntimeConfig` to run before `_BuildNativeLibrary` because the only dependency is that the actual file is in the "bundle_path" when the runtime is intialized.